2.4/contrib/bin/run.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# SCL in CentOS/RHEL 7 doesn't support --exec, we need to do it ourselves
+source scl_source enable mongodb24
+set -e
+
+function usage {
+	echo "You must specify following environment variables:"
+	echo "  \$MONGODB_USER"
+	echo "  \$MONGODB_PASSWORD"
+	echo "  \$MONGODB_DB - optional (default database - 'production')"
+	echo "  \$MONGODB_ADMIN_PASSWORD - optional"
+	exit 1
+}
+
+function create_mongodb_users {
+	mongod -f /opt/openshift/etc/mongodb.conf &
+
+	# Check if the MongoDB daemon is up.
+	RC=1
+	while [[ RC -ne 0 ]]; do
+	    echo "=> Waiting for confirmation of MongoDB service startup"
+	    sleep 2
```

Comment:
given there are no external dependencies i'm not sure why we'd have any hope we'd get different behavior if we relaunch it...  if we do, that's bad.  

also we do have health/liveness checks which (if properly used) would catch the fact that this container is not in fact healthy though it is running.

The main argument I can see for having a timeout on these loops is because otherwise a user might not be sure how long they should reasonably wait for it to start.  By us setting a timeout we're explicilty telling them "if it hasn't started yet, it's not going to".  I might be convinced by that argument.